pandas/core/internals.py

```diff
@@ -2753,6 +2761,18 @@ def _astype(self, dtype, copy=False, raise_on_error=True, values=None,
         return self.make_block_same_class(values=values,
                                           placement=self.mgr_locs)
 
+    def _can_hold_element(self, element):
+        return np.can_cast(np.asarray(element).dtype, self.sp_values.dtype)
```

Comment:
This seems like a reasonable default. @jreback your recent commit https://github.com/pandas-dev/pandas/commit/4efe6560e07f28de6a1834fa90e31cef31b0fb18#diff-e705e723b2d6e7c0e2a0443f80916abfR609 indicates this must be strict(er)?

Reply:
yes this looks ok